osu.Game/Screens/Ranking/RetryButton.cs

```diff
@@ -48,7 +46,14 @@ private void load(OsuColour colours)
             background.Colour = colours.Green;
 
             if (player != null)
+            {
+                TooltipText = player is ReplayPlayer ? "replay" : "retry";
                 Action = () => player.Restart();
+            }
+            else
+            {
+                TooltipText = "retry";
+            }
```

Comment:
Could keep the thing in its entirety within the ctor. In-fact I would probably move the `Action` binding there too.

This is because the `player is ReplayPlayer` line results in "retry" anyway. That said, the existence of this code block leads me to believe the `else` condition wasn't supposed to be "retry" too?

Reply:
It's here because of DI requirement for `Player`.

The else condition is just to show the tooltip when in a disabled state.